contracts/src/utils/structs/bitmap.rs

```diff
@@ -27,66 +27,90 @@ sol_storage! {
     }
 }
 
-impl BitMap {
+/// Required interface of a [`BitMap`] compliant contract.
+pub trait IBitMap {
```

Comment:
Do you think we need that kind of abstraction for `BitMap` and `Trace`?
I considering them mostly as primitives not separate contracts..
cc @ggonzalez94

Reply:
I would say that everything under `contracts` should have trait.

Reply:
May every function that is assumed to be publicly exposed should be in the trait? (public in solidity) In specified "contracts" there is no function that is assumed to be exposed publicly.

Reply:
Curious about @ggonzalez94 thoughts.

Reply:
I don't have a strong opinion here.... I don't think many people would use the trait and will only be interested in the implementation, but I guess for consistency it doesn't hurt to have them

Reply:
Since OL shared that they're going to stick to trait based approach.
And that in future we are expecting being able to declare `#[public]` macro more then once.
<img width="605" alt="image" src="https://github.com/user-attachments/assets/870e73ef-d0e0-4749-88bd-53bdcc4e27b3">
So this pattern can be done for every trait we have. That is why I'm considering to put just public methods into it.